pkg/resource/statefulset.go

```diff
@@ -191,6 +194,8 @@ func (b StatefulSetBuilder) makePodTemplate() corev1.PodTemplateSpec {
 	pod := corev1.PodTemplateSpec{
 		ObjectMeta: metav1.ObjectMeta{
 			Labels: b.Selector,
```

Comment:
Does this cause pod restart? Does datadog, for instance, need annotations on the pods?

Reply:
Yes, it does. Change it sts spec triggers restart.
Also, most of the monitoring tools like Datadog, Prometheus works on pod annotations only, so added custom annotations in the pod meta as well